src/script/sconsign.py

```diff
@@ -270,19 +277,19 @@ def map_timestamp(entry, name):
     else:
         return str(timestamp)
 
-def map_bkids(entry, name):
+def map_bkids(entry, _):
     try:
         bkids = entry.bsources + entry.bdepends + entry.bimplicit
         bkidsigs = entry.bsourcesigs + entry.bdependsigs + entry.bimplicitsigs
     except AttributeError:
         return None
-    result = []
-    for i in range(len(bkids)):
-        result.append(nodeinfo_string(bkids[i], bkidsigs[i], "        "))
+    result = [nodeinfo_string(bkid, bkidsig, "        ")
```

Comment:
This problem turned up after a complex build, sconsign file is 2404476 bytes. thows an IndexError because `bkidsigs` is zero length, so indexing into it inside the loop failed.  There are multiple instances in our build of this problem seen if we cause it not to error, but all come from the same sconscript/environment.  This change causes sconsign not to error out but does not address why signatures are missing - it follows the manpage comment of "the lines are simply omitted" since in these cases the function will return an empty list aka None.